File: iosApp/iosApp/widgets/DatePicker.swift
import SwiftUI

func openCalendar(
    onValueChanged: @escaping (Date) -> Void,
    onCancelled: @escaping () -> Void = {}
) {
    let calendar = Calendar.current
    let currentYear = calendar.component(.year, from: Date())
    let currentMonth = calendar.component(.month, from: Date())
    let currentDay = calendar.component(.day, from: Date())
    let currentHour = calendar.component(.hour, from: Date())
    let currentMinute = calendar.component(.minute, from: Date())

    let dialog = DatePicker("Select a date", selection: .constant(calendar.date(from: DateComponents(year: currentYear, month: currentMonth, day: currentDay, hour: currentHour, minute: currentMinute)) ?? Date()), displayedComponents: [.date, .hourAndMinute])
        dialog.datePickerStyle(.wheel)
//        dialog.onChange(of: dialog.selection) { date in
//            onValueChanged(date)
//            onCancelled()
//        }
        dialog.onDisappear {
            onCancelled()
        }
}

File: androidApp/src/main/java/pl/msiwak/multiplatform/android/ui/widgets/DatePicker.kt
package pl.msiwak.multiplatform.android.ui.widgets

import android.app.DatePickerDialog
import android.content.Context
import java.util.Calendar
import java.util.Date
import kotlinx.datetime.LocalDateTime

fun openCalendar(
    context: Context,
    onValueChanged: (LocalDateTime) -> Unit,
    onCancelled: () -> Unit = {}
) {
    val calendar = Calendar.getInstance()
    val currentYear = calendar.get(Calendar.YEAR)
    val currentMonth = calendar.get(Calendar.MONTH)
    val currentDay = calendar.get(Calendar.DAY_OF_MONTH)
    val currentHour = calendar.get(Calendar.HOUR_OF_DAY)
    val currentMinute = calendar.get(Calendar.MINUTE)

    val dialog = DatePickerDialog(context, { _, year, month, day ->
        onValueChanged(LocalDateTime(year, month + 1, day, currentHour, currentMinute))
        onCancelled()
    }, currentYear, currentMonth, currentDay)
    dialog.datePicker.maxDate = Date().time
    dialog.setOnCancelListener {
        onCancelled()
    }
    dialog.show()
}